### ISSUE TITLE
Silverstripe 4.7 PHP 500 error - MacOS PHP7.4.9

### ISSUE THREAD

**Issue by mr-macedawg** [silverstripe/silverstripe-mssql#58]
[29-Dec-2020 10:06:50 Australia/Brisbane] PHP Fatal error:  Uncaught Symfony\Component\Yaml\Exception\ParseException: The reserved indicator "%" cannot start a plain scalar; you need to quote the scalar at line 6 (near "connector: %$PDOConnector"). in /Applications/MAMP/htdocs/ss/vendor/symfony/yaml/Inline.php:305
Stack trace:
#0 /Applications/MAMP/htdocs/ss/vendor/symfony/yaml/Inline.php(89): Symfony\Component\Yaml\Inline::parseScalar('%$PDOConnector', 0, NULL, 14, true, Array)
#1 /Applications/MAMP/htdocs/ss/vendor/symfony/yaml/Parser.php(793): Symfony\Component\Yaml\Inline::parse('%$PDOConnector', 0, Array)
#2 /Applications/MAMP/htdocs/ss/vendor/symfony/yaml/Parser.php(339): Symfony\Component\Yaml\Parser->parseValue('%$PDOConnector', 0, 'mapping')
#3 /Applications/MAMP/htdocs/ss/vendor/symfony/yaml/Parser.php(521): Symfony\Component\Yaml\Parser->doParse('connector: %$PD...', 0)
#4 /Applications/MAMP/htdocs/ss/vendor/symfony/yaml/Parser.php(317): Symfony\Component\Yaml\Parser->parseBlock(5, 'connector: %$PD...', 0)
#5 /Applications/MAMP/htdocs/ss/vendor/symfony/yaml in /Applications/MAMP/htdocs/ss/vendor/symfony/yaml/Inline.php on line 305

**Comment by martinduparc** [comment]
It seems to be coming from the mysite.yml config file and the use of https://docs.silverstripe.org/en/4/developer_guides/debugging/error_handling/#sending-emails

It seems that using single quotes around the values where `%` is used solves the issue.

```yml
SilverStripe\Core\Injector\Injector:
  Psr\Log\LoggerInterface:
    calls:
      MailHandler: [ pushHandler, [ '%$MailHandler' ] ]
  MailHandler:
      class: Monolog\Handler\NativeMailerHandler
      constructor:
      properties:
        ContentType: text/html
        Formatter: '%$SilverStripe\Logging\DetailedErrorFormatter'
  SilverStripe\Security\MemberAuthenticator\MemberAuthenticator:
    class: ECMemberAuthenticator
```

If someone can confirm the fix, I can create a PR to update the documentation.

**Comment by mr-macedawg** [comment]
See this: https://docs.silverstripe.org/en/4/changelogs/4.7.0/#support-for-symfony-4-components

Manually adding single quotes in the connectors.yml rectified the issue.

**Comment by brynwhyman** [comment]
Yes, this is due to the automatic upgrade to Symfony 4 in CMS 4.7. It looks like there are a couple of YML files in this module that need to be updated for compatibility (quoting strings starting with %). Would you like to contribute with a pull request to help out here @mr-macedawg?

**Comment by mr-macedawg** [comment]
Already have @brynwhyman. Just waiting for it to be approved #59

**Comment by mr-macedawg** [comment]
@brynwhyman I have changed the PR to target branch 2.0 #60

**Comment by brynwhyman** [comment]
Closed with #60, thank you!
